### PR TITLE
Add append-dropdown-to-body option

### PR DESCRIPTION
**Pull request by lameyer**
This adds an option to ui-select to append only the dropdown to the body (instead of the entire select box as happens with `append-to-body`). Adding `append-dropdown-to-body` to the `ui-select` element will append both the `ui-select-choices` and `ui-select-no-choices` elements to the body.

Generally, this option will make ui-select work better within scrolling divs and modals.

This also fixes a bug with `tag-on-blur` when it was used with `append-to-body`.

Here's a plunk for testing!
http://run.plnkr.co/plunks/NZUTFvsONvsrPrwoQez9/

### CODE REVIEW

**Comment by lameyer** [comment]
cc @whscullin

File: src/common.css [hunk 1]
```diff
@@ -190,7 +194,8 @@ body > .select2-container.open {
 }
 
 /* See Scrollable Menu with Bootstrap 3 http://stackoverflow.com/questions/19227496 */
-.ui-select-bootstrap > .ui-select-choices ,.ui-select-bootstrap > .ui-select-no-choice {
+.ui-select-container > .ui-select-choices-bootstrap,
```
Comment (arkarkark) [inline src/common.css:197]: this is going to break the non bootstrap usage of ui-select, right? so at this point we're admitting we're never going to try and get this stuff pushed upstream?

File: src/uiSelectDirective.js [hunk 2]
```diff
@@ -171,6 +171,12 @@ uis.directive('uiSelect',
           $select.spinnerClass = spinnerClass !== undefined ? attrs.spinnerClass : uiSelectConfig.spinnerClass;
         });
 
+        attrs.$observe('appendDropdownToBody', function() {
+          // $eval() is needed otherwise we get a string instead of a boolean
+          var appendDropdownToBody = scope.$eval(attrs.appendDropdownToBody);
+          $select.appendDropdownToBody = appendDropdownToBody !== undefined ? appendDropdownToBody : uiSelectConfig.appendDropdownToBody;
+        });
```
Comment (arkarkark) [inline src/uiSelectDirective.js:178]: do you need to worry about this changing while something is UP and appendedToBody and  now you need to reparent it? my guess is no.

File: src/uiSelectDirective.js [hunk 3]
```diff
@@ -428,6 +442,101 @@ uis.directive('uiSelect',
             element.removeClass(directionUpClassName);
           }
         };
+
+        // Support for appending just the dropdown to the body when it's open
+        scope.$watch('$select.open', function() {
+          positionOnlyDropdown();
```
Comment (arkarkark) [inline src/uiSelectDirective.js:448]: does this need to be wrapped in a `if ($select.appendDropdownToBody) {` ?

File: src/uiSelectDirective.js [hunk 4]
```diff
@@ -428,6 +442,101 @@ uis.directive('uiSelect',
...
+
+        // Close the dropdown if the scrolling parent is scrolled
+        // since the appended dropdown will not scroll with the container
+        var scrollParent = angular.element(getScrollParent(element[0]));
+        if (scrollParent) {
+          scrollParent.on('scroll', function() {
+            if ($select.appendDropdownToBody && $select.open) {
+              $timeout(function() {
+                $select.close($select.skipFocusser);
+              },0);
```
Comment (arkarkark) [inline src/uiSelectDirective.js]: no need for the `,0` if you don't want on $timeout

**Review by arkarkark — COMMENTED** [review]
looks fine as much as I understand this stuff.

I really think a lot of this would be great going upstream.

File: src/uiSelectDirective.js (hunk 3 above)
Comment (lameyer) [reply src/uiSelectDirective.js:448]: I do that within the `positionOnlyDropdown` method!

File: src/common.css (hunk 1 above)
Comment (lameyer) [reply src/common.css:197]: The select2 and selectize themes don't do styling with the same sort of classes - as far as I can tell, the css for those shouldn't be affected by this. I did some testing with the other themes as well, and they seemed to work with append-dropdown-to-body!

**Comment by arkarkark** [comment]
oh and you tested on IE and safari and firefox? this seems like one of those things that might trip up on those?

File: src/uiSelectDirective.js (hunk 2 above)
Comment (lameyer) [reply src/uiSelectDirective.js:178]: Not sure what you mean?

**Comment by lameyer** [comment]
Great thing you asked, because I tested before but something got changed so it didn't work in IE11! Got a quick fix for that 👍

File: src/uiSelectDirective.js (hunk 2 above)
Comment (arkarkark) [reply src/uiSelectDirective.js:178]: you're using $eval so you could have append-dropdown-to-body="controller.thing"

oh wait... never mind...

sorry. pls ignore... :-D

**Pull request merged**